tests/operator_test.go

```diff
@@ -1373,6 +1347,112 @@ var _ = Describe("ALL Operator tests", func() {
 	})
 })
 
+func getCDICrAndPods(f *framework.Framework) (*cdiv1.CDI, *corev1.PodList) {
+	cdiPods, err := f.K8sClient.CoreV1().Pods(f.CdiInstallNs).List(context.TODO(), metav1.ListOptions{})
+
+	Expect(err).ToNot(HaveOccurred(), "failed listing cdi pods")
+	Expect(len(cdiPods.Items)).To(BeNumerically(">", 0), "no cdi pods found")
+
+	cr, err := f.CdiClient.CdiV1beta1().CDIs().Get(context.TODO(), "cdi", metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		Skip("CDI CR 'cdi' does not exist.  Probably managed by another operator so skipping.")
+	}
+	Expect(err).ToNot(HaveOccurred())
+	return cr, cdiPods
+}
+
+func removeCDI(f *framework.Framework, cr *cdiv1.CDI) {
+	By("Deleting CDI CR if exists")
+	_ = f.CdiClient.CdiV1beta1().CDIs().Delete(context.TODO(), cr.Name, metav1.DeleteOptions{})
+
+	By("Waiting for CDI CR and infra deployments to be gone now that we are sure there's no CDI CR")
+	Eventually(func() bool { return infraDeploymentGone(f) && crGone(f, cr) }, 15*time.Minute, 2*time.Second).Should(BeTrue())
+}
+
+func ensureCDI(f *framework.Framework, cr *cdiv1.CDI, cdiPods *corev1.PodList) {
+	if cr == nil {
+		return
+	}
+
+	By("Check if CDI CR exists")
+	cdi, err := f.CdiClient.CdiV1beta1().CDIs().Get(context.TODO(), cr.Name, metav1.GetOptions{})
+	if err == nil {
+		if cdi.DeletionTimestamp == nil {
+			By("CDI CR exists")
+			cdi.Spec = cr.Spec
+			_, err = f.CdiClient.CdiV1beta1().CDIs().Update(context.TODO(), cdi, metav1.UpdateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			return
+		}
+
+		By("Waiting for CDI CR deletion")
+		Eventually(func() bool {
+			_, err = f.CdiClient.CdiV1beta1().CDIs().Get(context.TODO(), cr.Name, metav1.GetOptions{})
+			if errors.IsNotFound(err) {
+				return true
+			}
+			Expect(err).ToNot(HaveOccurred())
+			return false
+		}, 5*time.Minute, 2*time.Second).Should(BeTrue())
+	} else {
+		Expect(errors.IsNotFound(err)).To(BeTrue())
+	}
+
+	cdi = &cdiv1.CDI{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cdi",
```

Comment:
Same here, the name of the CR is hard coded to `cdi`, we should take the name from the passed in CR